stl/inc/expected

```diff
@@ -0,0 +1,1034 @@
+    bad_expected_access& operator=(const bad_expected_access&) = default;
+    bad_expected_access& operator=(bad_expected_access&&) = default;
+
+#if !_HAS_EXCEPTIONS
+    void _Doraise() const override { // perform class-specific exception handling
+        _RAISE(*this);
+    }
+#endif // !_HAS_EXCEPTIONS
+};
+
+template <class _Err>
+class bad_expected_access : public bad_expected_access<void> {
+public:
+    explicit bad_expected_access(_Err _Unex) noexcept(is_nothrow_move_constructible_v<_Err>) // strengthened
+        : _Unexpected(_STD move(_Unex)) {}
+
+#ifdef __cpp_explicit_this_parameter // TRANSITION, P2549, P0847R7
+    _NODISCARD decltype(auto) error(this auto&& _Self) noexcept {
+        return (_STD forward<decltype(_Self)>(_Self)._Unexpected);
+    }
+#else // ^^^ __cpp_explicit_this_parameter ^^^ / vvv !__cpp_explicit_this_parameter vvv
+    _NODISCARD const _Err& error() const& noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD _Err& error() & noexcept {
+        return _Unexpected;
+    }
+    _NODISCARD const _Err&& error() const&& noexcept {
+        return _STD move(_Unexpected);
+    }
+    _NODISCARD _Err&& error() && noexcept {
+        return _STD move(_Unexpected);
+    }
+#endif // !__cpp_explicit_this_parameter
+
+private:
+    _Err _Unexpected;
+};
+
+struct unexpect_t {
+    explicit unexpect_t() = default;
+};
+
+inline constexpr unexpect_t unexpect{};
+
+template <class _Ty, class _Err>
+class expected {
+    template <class _UTy, class _UErr>
+    friend class expected;
+
+public:
+    using value_type      = _Ty;
+    using error_type      = _Err;
+    using unexpected_type = unexpected<_Err>;
+
+    template <class _Uty>
+    using rebind = expected<_Uty, error_type>;
+
+    // [expected.object.ctor]
+    constexpr expected() noexcept(is_nothrow_default_constructible_v<_Ty>) // strengthened
+        requires is_default_constructible_v<_Ty> //
+        : _Value(), _Has_value(true) {}
+
+    // clang-format off
+    constexpr expected(const expected& _Other) noexcept(
+        is_nothrow_copy_constructible_v<_Ty>&& is_nothrow_copy_constructible_v<_Err>) // strengthened
+        requires (!is_trivially_copy_constructible_v<_Ty> && !is_trivially_copy_constructible_v<_Err>
+                 && is_copy_constructible_v<_Ty> && is_copy_constructible_v<_Err>)
+        : _Has_value(_Other._Has_value) {
+        // clang-format on
+        if (_Has_value) {
+            _STD construct_at(_STD addressof(_Value), _Other._Value);
+        } else {
+            _STD construct_at(_STD addressof(_Unexpected), _Other._Unexpected);
+        }
+    }
+
+    // clang-format off
+    expected(const expected& _Other) requires
+        is_trivially_copy_constructible_v<_Ty> && is_trivially_copy_constructible_v<_Err> = default;
+
+    constexpr expected(expected&& _Other) noexcept(
+        is_nothrow_move_constructible_v<_Ty>&& is_nothrow_move_constructible_v<_Err>) // strengthened

```

Comment:
@CaseyCarter noticed that this exception specification is mandated by the Standard and therefore should not be commented as `strengthened`.